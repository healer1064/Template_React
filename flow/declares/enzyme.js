// @flow

declare module 'enzyme' {

  declare var shallow: Function

}
