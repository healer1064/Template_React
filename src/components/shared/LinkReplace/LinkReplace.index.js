// @flow
import LinkReplace from './LinkReplace'

export default LinkReplace
