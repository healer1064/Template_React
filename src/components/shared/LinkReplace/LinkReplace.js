// @flow
import React from 'react'
import { Link } from 'react-router'
import history from 'src/helpers/history'

type ReactProps = {
  to: Object,
  className?: any,
  children?: React$Element<any>,
};

type Props = ReactProps;

const LinkReplace = (props: Props): React$Element<any> => {

  return (
    <Link
      className={props.className}
      onClick={() => {

        if (props.to.state && props.to.query) {

          history.replace({
            pathname: props.to.pathname,
            state: props.to.state,
            query: props.to.query,
          })

        }

        else if (props.to.state) {

          history.replace({
            pathname: props.to.pathname,
            state: props.to.state,
          })

        }

        else if (props.to.query) {

          history.replace({
            pathname: props.to.pathname,
            query: props.to.query,
          })

        }

        else {

          history.replace({
            pathname: props.to,
            state: { modal: '' },
          })

        }

      }
    }

    >
      {props.children}
    </Link>
  )

}

export default LinkReplace
