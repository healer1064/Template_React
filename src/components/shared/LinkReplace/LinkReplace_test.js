// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import LinkReplace from './LinkReplace'


it('<LinkReplace> renders the LinkReplace component', () => {

  const wrapper = shallow(
    <LinkReplace
      to={{
        pathname: 'location_string',
        state: { modal: 'create-offer-intro' },
      }}
    >
      <button />
    </LinkReplace>
  )

  expect(wrapper).toMatchSnapshot()

})
