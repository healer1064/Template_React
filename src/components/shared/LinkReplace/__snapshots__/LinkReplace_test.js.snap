// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkReplace> renders the LinkReplace component 1`] = `
<Link
  onClick={[Function]}
  onlyActiveOnIndex={false}
  style={Object {}}
>
  <button />
</Link>
`;
